test/code/scxsystemlib/disk/scxlvmutils_test.cpp

```diff
@@ -1036,6 +1039,143 @@ class SCXLVMUtilsTest : public CPPUNIT_NS::TestFixture
 
         CPPUNIT_ASSERT_EQUAL(((size_t) 3), index);
     }
+
+    /**
+     * This class mocks SCXLVMUtilsDepends for testing GetDMSlaves.
+     * Given a map M of strings to vectors of strings, this class simulates the following directory structure
+     * for each entry (X, [Y1,Y2,..Yn]) of M:
+     * /sys/block/X
+     * /sys/block/X/dev  <-- contains one line "253:2"
+     * /sys/block/X/slaves
+     * /sys/block/X/slaves/Y1
+     * /sys/block/X/slaves/Y2
+     * ....
+     * /sys/block/X/slaves/Yn
+     */
+    class TestGetDMSlavesSCXLVMUtilDepends
+        : public SCXSystemLib::SCXLVMUtilsDepends
+    {
+    public:
+        std::map< std::wstring, std::vector<std::wstring> > & m_slaves;
+
+        TestGetDMSlavesSCXLVMUtilDepends(std::map< std::wstring, std::vector< std::wstring > > & slaves) : m_slaves(slaves) { }
+
+        virtual std::vector< SCXCoreLib::SCXFilePath > GetFileSystemEntries(
+            const SCXCoreLib::SCXFilePath               & path,
+            const SCXCoreLib::SCXDirectorySearchOptions  options)
+        {
+            std::wstring slavesExt = L"/slaves/";
+            std::wstring pathStr = path.Get().substr(0, path.Get().length() - slavesExt.length());
+            SCXCoreLib::SCXFilePath blockFile = pathStr;
+
+            // to avoid getting unused variable warning
+            (void) options;
+
+            std::vector< SCXCoreLib::SCXFilePath > result;
+            std::vector<std::wstring> slaveEntries = m_slaves[blockFile.GetFilename()];
+
+            for (std::vector< std::wstring >::const_iterator iter = slaveEntries.begin(); iter != slaveEntries.end(); iter++)
+            {
+                SCXCoreLib::SCXFilePath slavePath(path);
+
+                slavePath.AppendDirectory(*iter);
+                result.push_back(slavePath);
+            }
+            return (result);
+        }
+
+
+        virtual void Stat(
+            const SCXCoreLib::SCXFilePath            & path,
+            SCXCoreLib::SCXFileSystem::SCXStatStruct * pStat)
+        {
+            // warnings as errors, so deal with the unused params
+            (void) path;
+            pStat->st_rdev = makedev(253, 2);
+        }
+
+
+        virtual void ReadAllLinesAsUTF8(
+            const SCXCoreLib::SCXFilePath & source,
+            std::vector< std::wstring >   & lines,
+            SCXCoreLib::SCXStream::NLFs   & nlfs)
+        {
+            // warnings as errors, so deal with the unused params
+            (void) source;
+            (void) nlfs;
+
+            lines.clear();
+            lines.push_back(L"253:2");
+
+        }
+
+        virtual ~TestGetDMSlavesSCXLVMUtilDepends() { }
+    };
+
+    void GetDMSlaves_SlavesWithDMEntries_TraversesToDevice()
+    {
+
```

Comment:
Delete needless blank line after {. Blank lines are inserted for readability, in this case it's not all that useful.